controller/api/public/prometheus.go

```diff
@@ -116,6 +118,20 @@ func promDstQueryLabels(resource *pb.Resource) model.LabelSet {
 	return set
 }
 
+// insert a not-nil check into a LabelSet to verify that data for a specified
+// label name exists. due to the `!=` this must be inserted as a string. the
+// structure of this code is taken from the Prometheus labelset.go library.
+func customToString(l model.LabelSet, labelName string) string {
```

Comment:
nit: I think it should also be made clearer from the name of this fn that it's generating the prometheus label query string with exclusions. Wordier but more descriptive suggestion: `generateLabelStringWithExclusion`

Reply:
How about `generateLabelSetStringWithNotNilCheck`?

Reply:
fine by me!